Assets/HexMap.cs

```diff
@@ -226,96 +201,54 @@ public Hex GetAcceptableStartPosition()
 
         return possibleHex;
     }
-
-    public GameObject GetHexGOFromHex(Hex hex)
-    {
-        GameObject hexGO = hexToHexGOMap[hex];
-        return hexGO;
-    }
     
-    public void UpdateHexVisuals()
+    public Hex[] GetHexesWithRadiusOf(Hex centerHex, int radius)
     {
-        for (int column = 0; column < NumColumns(); column++)
+        List<Hex> results = new List<Hex>();
+        for (int dx = -radius; dx < radius; dx++)
         {
-            for (int row = 0; row < NumRows(); row++)
+            for (int dy = Mathf.Max(-radius, -dx - radius); dy < Mathf.Min(radius, -dx + radius); dy++)
```

Comment:
Nested for loop, complicated calculation with no comment (again I'd look at a dedicated calculation solution).